server/modules/express-middleware/deprecateMiddleware.js

```diff
@@ -0,0 +1,26 @@
+/* Copyright (c) Trainline Limited, 2016-2017. All rights reserved. See LICENSE.txt in the project root for license information. */
+
+/**
+ * Express middleware that adds deprecated=true to a request and sets the Warning HTTP header
+ * on the response if it is a request to a deprecated route.
+ */
+
+'use strict';
+
+function create(fn) {
+  return function deprecateMiddleware(req, res, next) {
+    try {
+      let warning = fn(req);
+      if (warning) {
+        let now = new Date().toUTCString();
+        res.locals.deprecated = true;
+        res.append('Warning', `299 - Deprecated: ${warning} "${now}"`);
```

Comment:
I like a 301 here. I don't know how clients would react to the 3xx, and whether they would get their knickers in a twist at that response... but I think it's correct.